soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +226,57 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
     }
 }
 
+bool EnCow_IsSecondCowInTower(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_SOUKO && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -173 || this->actor.world.pos.x + this->actor.world.pos.z == -72);
+}
+
+bool EnCow_IsSecondCowInStable(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_MALON_STABLE && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -257 || this->actor.world.pos.x + this->actor.world.pos.z == -138);
+}
+
+bool EnCow_IsSecondCowInGrotto(EnCow* this, GlobalContext* globalCtx) {
+    return gSaveContext.n64ddFlag && globalCtx->sceneNum == SCENE_KAKUSIANA && this->actor.world.pos.x + this->actor.world.pos.z == 3194;
+}
+
+bool EnCow_IsSecondCow(EnCow* this, GlobalContext* globalCtx) {
+    return EnCow_IsSecondCowInTower(this, globalCtx) || EnCow_IsSecondCowInStable(this, globalCtx) || EnCow_IsSecondCowInGrotto(this, globalCtx);
+}
+
+GetItemID EnCow_GetItemFromCow(EnCow* this, GlobalContext* globalCtx, bool setFlag) {
+    GetItemID itemId = ITEM_NONE;
+
+    if (Inventory_HasEmptyBottle()) {
+        return GI_MILK;
+    }
+
+    s32 collectableFlag = 0x1A;
+    if (EnCow_IsSecondCow(this, globalCtx)) {
+        collectableFlag = 0x19;
+    }
+
+    if (gSaveContext.n64ddFlag && !Flags_GetCollectible(globalCtx, collectableFlag)) {
+        itemId = Randomizer_GetRandomizedItemId(GI_MILK, this->actor.id, this->actor.world.pos.x + this->actor.world.pos.z, globalCtx->sceneNum);
+    }
+
+    if (setFlag) {
+        Flags_SetCollectible(globalCtx, collectableFlag);
+    }
+
+    return itemId;
+}
+
 void func_809DF778(EnCow* this, GlobalContext* globalCtx) {
     if (Actor_HasParent(&this->actor, globalCtx)) {
         this->actor.parent = NULL;
         this->actionFunc = func_809DF730;
     } else {
-        func_8002F434(&this->actor, globalCtx, GI_MILK, 10000.0f, 100.0f);
+        func_8002F434(&this->actor, globalCtx, EnCow_GetItemFromCow(this, globalCtx, true), 10000.0f, 100.0f);
```

Comment:
not sure how i feel about having the 64dd check happening instead `GetItemFromCow`, i think just having
```c
func_8002F434(&this->actor, globalCtx, gSaveContext.n64ddFlag ? EnCow_GetItemFromCow(this, globalCtx, true) : GI_MILK, 10000.0f, 100.0f);
```
might be better

i also feel like it may make sense to move `GetItemFromCow` out to randomizer.cpp (with the intention of splitting randomizer.cpp up in the future) since it should only be used when we're rando'd